src/vector/platform/tchap-desktop/TauriSecureStorage.ts

```diff
@@ -0,0 +1,80 @@
+import { Client, Store, Stronghold } from '@tauri-apps/plugin-stronghold';
+import { appDataDir } from '@tauri-apps/api/path';
+import { logger } from 'matrix-js-sdk/src/logger';
+
+// Tauri secure storage - using strongholg
+export class TauriSecureStorage {
+    private store: Store | undefined;
+    private client: Stronghold | undefined;
+
+    public constructor() {
+    }
+
+    public getRandom32Bytes(): Uint8Array {
+        return crypto.getRandomValues(new Uint8Array(32));
+    }
+
+    public getRandomUtf832Bytes(): Uint8Array{
+        // Define the Unicode range for Greek alphabet characters
+        const minCodePoint = 0x0391; // 'Α' (Greek Capital Letter Alpha)
+        const maxCodePoint = 0x03A9; // 'Ω' (Greek Capital Letter Omega)
+        const range = maxCodePoint - minCodePoint + 1;
+
+        // Create a Uint32Array to hold the random value
+        const randomValue = new Uint8Array(32);
+        crypto.getRandomValues(randomValue);
+
+        // Map the random value to the desired Unicode range
+        const randomCodePoint = randomValue.map(value => minCodePoint + (value % range));
+
+        // Convert the code point to a character
+        return randomCodePoint;
+    }
+
+    public async getItem(key: string): Promise<any> {
+        if (!this.store) return Promise.reject(new Error("No stronghold store found"));
+        return this.store.get(key);
+    }
+
+    public async createItem(key: string, value: any): Promise<any> {
+        if (!this.store) return Promise.reject(new Error("No stronghold store found"));
+
+        this.store?.insert(key, Array.from(value));
+
+        // Save your updates
+        await this.client?.save();
+    }
+
+    public async removeItem(key: string): Promise<any> {
+        if (!this.store) return Promise.reject(new Error("No stronghold store found"));
+
+        return this.store?.remove(key);
+    }
+
+    // On dev build, loading the stronghold vault takes a really long time https://github.com/tauri-apps/tauri/issues/4197
+    public async initStronghold(): Promise<void> {
+        logger.log("[Tauri] Initializing stronghold");
+        try {
+            if (!this.store) {
+                const vaultPath = `${await appDataDir()}/vault.hold`;
+                logger.log("[Tauri] vaulpath", vaultPath);
+                const vaultPassword = 'tchap-desktop-vault321';
```

Comment:
We need to initialize a password for each client. This one will be the same across the clients, but will be salted with a unique salt for each installation https://github.com/tchapgouv/tchap-desktop/blob/41d4001e662543f8bfcf200463f7a727b9d60889/src-tauri/src/lib.rs#L39